kiota_abstractions/request_information.py

```diff
@@ -95,46 +96,25 @@ def url(self, url: Url) -> None:
     @property
     def request_headers(self) -> Optional[Dict]:
         final = {}
-        for key, value in self.headers.items():
-            final[key] = ", ".join(value)
+        for key, values in self.headers.get_all().items():
+            final[key] = ', '.join(values)
         return final
 
-    def add_request_headers(
-        self, headers_to_add: Optional[Dict[str, Union[str, List[str]]]]
-    ) -> None:
-        """Adds headers to the request"""
-        if headers_to_add:
-            for key, value in headers_to_add.items():
-                lowercase_key = key.lower()
-                if lowercase_key in self.headers:
-                    if isinstance(value, list):
-                        self.headers[lowercase_key] = self.headers[lowercase_key].union(set(value))
-                    else:
-                        self.headers[lowercase_key].add(str(value))
-                else:
-                    if isinstance(value, list):
-                        self.headers[lowercase_key] = set(value)
-                    else:
-                        self.headers[lowercase_key] = {str(value)}
-
-    def try_add_request_header(self, key: str, value: str) -> bool:
-        """Try to add an header to the request if it's not already set"""
-        if key and value:
-            lowercase_key = key.lower()
-            if lowercase_key in self.headers:
-                return False
-            self.headers[lowercase_key] = {str(value)}
-            return True
-        return False
+    def add_request_headers(self, headers_to_add: RequestHeaders) -> None:
```

Comment:
is there much value in keeping those two vanity methods? Python would be the only language with those.

Reply:
Not really needed. Removed.